### PR TITLE
Competition Auto-Run Submissions

### PR DESCRIPTION
**Pull request by ihsaan-ullah**
# @ mention of reviewers
@Didayolo 

# A brief description of the purpose of the changes contained in this PR.
Organizers can enable or disable `auto_run_submissions` feature from competition setting to allow participant's submissions to auto-run (the default way) or to require an organizer to run the submission manually (the new feature)

### How to set `auto_run_submissions` 
1. You can set it in the competition settings:
Screenshot 2024-01-23 at 11 07 12 AM

2. You can set it in the yaml file:
Screenshot 2024-01-23 at 11 07 37 AM

Note: default value is `True` so that the 

### How do participants know their submission are not auto-run?
Participants see a question mark with a message to tell them that organizers will run their submissions:
Screenshot 2024-01-23 at 11 10 38 AM

### How can organizer run the submission?
Organizer can see a `play` button for submissions which should be executed manually. By clicking this button, the submission will run as a normal submission
Screenshot 2024-01-23 at 11 11 19 AM

# Issues this PR resolves
- #1281 

# A checklist for hand testing
- [x] Create a competition and disable `auto_run_submissions` either from competition setting or from competition yaml
- [x] Submit a submission and check if you see a message that the submission will be run by organizer
- [x] As an organizer, check that you see the play button in front of the submission and submission runs when you click it
- [x] upload two competitions one with `auto_run_submission` set to true and other with false In the yaml file to see they are correctly shown in the competition setting
- [x] Enable/Disable `auto_run_submissions` from the setting to check that uploaded submissions follow the setting

# Checklist
- [x] Code review by me 
- [x] Hand tested by me 
- [x] I'm proud of my work
- [x] Code review by reviewer
- [x] Hand tested by reviewer
- [x] CircleCi tests are passing
- [ ] Ready to merge

### CODE REVIEW

**Comment by cjh1** [comment]
Quick question, what is the difference between "Submitting and Submitted". To me I would expect as submission to be in the  "Submitted" state after I have uploaded my submission?

File: src/apps/api/serializers/submissions.py [hunk 1]
```diff
@@ -41,6 +41,7 @@ class SubmissionSerializer(serializers.ModelSerializer):
     on_leaderboard = serializers.BooleanField(read_only=True)
     task = TaskSerializer()
     created_when = serializers.DateTimeField(format="%Y-%m-%d %H:%M")
+    auto_run = serializers.SerializerMethodField(read_only=True)
```
Comment (cjh1) [inline src/apps/api/serializers/submissions.py:44]: Why do we need to add `auto_run` to every submission? To me if you want to see if a submission should be auto run, you should check the associated competition?

File: src/apps/competitions/models.py [hunk 2]
```diff
@@ -538,6 +542,13 @@ def start(self, tasks=None):
         from .tasks import run_submission
         run_submission(self.pk, tasks=tasks)
 
+    def run(self):
+        # get tasks from the phase
+        tasks = self.phase.tasks.all()
+        # start submission providing the tasks
+        self.start(tasks=tasks)
+        return self
+
```
Comment (cjh1) [inline src/apps/competitions/models.py:551]: Why is this change needed?

**Comment by ihsaan-ullah** [comment]
> Quick question, what is the difference between "Submitting and Submitted". To me I would expect as submission to be in the "Submitted" state after I have uploaded my submission?

The first status is`Submitting` when you upload a submission. Submitted is used when submission is forwarded for running

File: src/apps/api/serializers/submissions.py (hunk 1 above)
Comment (ihsaan-ullah) [reply src/apps/api/serializers/submissions.py:44]: This is added so that it can be accessed easily because submission is not directly linked to a competition but through a phase

File: src/apps/competitions/models.py (hunk 2 above)
Comment (ihsaan-ullah) [reply src/apps/competitions/models.py:551]: this change is for running the submission inside run_submission function. self.start forwards the submission for running
